### DISCUSSION TITLE
Ability to have custom http request handler

### DISCUSSION THREAD

**Discussion by dzorya** [webui-dev/webui#478] [Q&A] [answered]
Is it possible to have a custom http request handler in addition to webui_set_file_handler?
So it would be possible to process all requests types like POST, DELETE, etc... ?
This way it should be possible to use other frameworks (HTMX as example) to render UI in addition to calling functions over websocket.

**Comment by AlbertShown** [comment]
Is POST, DELETE... are not routed to webui_set_file_handler right now?

**Comment by dzorya** [comment]
I thought this if:
https://github.com/webui-dev/webui/blob/ead99700e99d25500f8a4a4b5ca9efcc6f28b47d/src/webui.c#L7854
would prevent it.
I guess, I should test it.

пн, 26 авг. 2024 г., 22:10 Showns ***@***.***>:

**Reply by AlbertShown** [reply]
Probably you are right. You can try:

```c
if (strcmp(ri->request_method, "GET") == 0 || 
    strcmp(ri->request_method, "POST") == 0 ||
    strcmp(ri->request_method, "PUT") == 0 ||
    strcmp(ri->request_method, "DELETE") == 0 ||
    strcmp(ri->request_method, "PATCH") == 0)
{
    // ...
}
```

If this worked for you, and all the C examples still working, then you can create a PR. Thank you.

**Comment by dzorya** [comment]
But webui_set_file_handler doesn't receive method or reqest data just url.
We would need a new handler function for this, wouldn't we?

вт, 27 авг. 2024 г., 00:21 Showns ***@***.***>:

**Reply by AlbertShown** [reply, marked as answer]
Yes, only the URLs, I guess the solution is to use a custom web server (apache, python server, nginx...)
See this example: https://github.com/webui-dev/webui/tree/main/examples/C/custom_web_server

**Comment by dzorya** [comment]
Thank you! This is exactly what I need.
